deployment/v2/4_createRollup.ts

```diff
@@ -441,31 +441,35 @@ async function main() {
         if (consensusContract == utilsAggchain.AGGCHAIN_CONTRACT_NAMES.ECDSA) {
             aggchainType = utilsECDSA.AGGCHAIN_TYPE_ECDSA;
         }
+
         // Load aggLayerGateway
         const aggLayerGatewayFactory = await ethers.getContractFactory("AggLayerGateway", deployer);
         const aggLayerGateway = (await aggLayerGatewayFactory.attach(
             deployOutput.aggLayerGatewayAddress
         )) as AggLayerGateway;
 
-        const defaultAggchainSelector = utilsAggchain.getAggchainVKeySelector(
-            createRollupParameters.aggchainParams.initAggchainVKeyVersion,
-            aggchainType
-        );
+        // sanity check on the aggchainType
+        const aggchainTypeParam = utilsAggchain.getAggchainTypeFromSelector(createRollupParameters.aggchainParams.initAggchainVKeySelector);
+
+        if (aggchainType !== aggchainTypeParam) {
+            console.error(`Aggchain type ${aggchainType} does not match the selector ${createRollupParameters.aggchainParams.initAggchainVKeySelector}`);
+            process.exit(0);
```

Comment:
```suggestion
            process.exit(1);
```

Reply:
Or better just throw error with the error message and it will be catched by the error handler (try catch)